src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -7574,6 +7574,63 @@ public enum NormalizationForm
         FormKC = 5,
         FormKD = 6,
     }
+    public readonly struct Rune : IComparable<Rune>, IEquatable<Rune>
+    {
+        private readonly int _dummyPrimitive;
+        public Rune(char ch) { throw null; }
+        public Rune(int value) { throw null; }
+        [CLSCompliant(false)]
+        public Rune(uint value) { throw null; }
+        public static bool operator ==(Rune left, Rune right) { throw null; }
+        public static bool operator !=(Rune left, Rune right) { throw null; }
+        public static bool operator <(Rune left, Rune right) { throw null; }
+        public static bool operator <=(Rune left, Rune right) { throw null; }
+        public static bool operator >(Rune left, Rune right) { throw null; }
+        public static bool operator >=(Rune left, Rune right) { throw null; }
+        public static explicit operator Rune(char ch) { throw null; }
+        [CLSCompliant(false)]
+        public static explicit operator Rune(uint value) { throw null; }
+        public static explicit operator Rune(int value) { throw null; }
+        public bool IsAscii { get { throw null; } }
+        public bool IsBmp { get { throw null; } }
+        public int Plane { get { throw null; } }
+        public static Rune ReplacementChar { get { throw null; } }
+        public int Utf16SequenceLength { get { throw null; } }
+        public int Utf8SequenceLength { get { throw null; } }
+        public int Value { get { throw null; } }
+        public int CompareTo(Rune other) { throw null; }
+        public override bool Equals(object obj) { throw null; }
+        public bool Equals(Rune other) { throw null; }
+        public override int GetHashCode() { throw null; }
+        public static Rune GetRuneAt(string input, int index) { throw null; }
+        public static bool IsValid(int value) { throw null; }
+        [CLSCompliant(false)]
+        public static bool IsValid(uint value) { throw null; }
+        public override string ToString() { throw null; }
+        public static bool TryCreate(char ch, out Rune result) { throw null; }
+        public static bool TryCreate(int value, out Rune result) { throw null; }
+        [CLSCompliant(false)]
+        public static bool TryCreate(uint value, out Rune result) { throw null; }
+        public bool TryEncode(Span<char> destination, out int charsWritten) { throw null; }
+        public static bool TryGetRuneAt(string input, int index, out Rune value) { throw null; }
+        public static double GetNumericValue(Rune value) { throw null; }
```

Comment:
Does returning `double` enable any specific use-cases that e.g. `long` does not?

Reply:
Unicode defines fractional values, such as U+00BD (½) for which a non-integral value is returned.

http://unicodefractions.com/ for some others.

Reply:
We pull the data from https://www.unicode.org/Public/UCD/latest/ucd/extracted/DerivedNumericValues.txt.
Our test file has a specific test for the U+00BD case:

```cs
yield return new object[] { new UnicodeInfoTestData { ScalarValue = (Rune)(0xBD), UnicodeCategory = UnicodeCategory.OtherNumber, NumericValue = 0.5, IsControl = false, IsDigit = false, IsLetter = false, IsLetterOrDigit = false, IsLower = false, IsNumber = true, IsPunctuation = false, IsSeparator = false, IsSymbol = false, IsUpper = false, IsWhiteSpace = false } };
```

Reply:
It gets even more fun, since some scalars can correspond to _negative_ numeric values! :D
For example, U+0F33 TIBETAN DIGIT HALF ZERO ('༳') has a numeric value of `-0.5`.

Reply:
Interesting. Is this API as generally useful as to be exposed directly on the `Rune` type?

Reply:
To be honest I don't personally have a valid use case for it. It's there for parity with `System.Char` and `System.Text.CharUnicodeInfo`.